packages/governance/src/types.js

```diff
@@ -87,79 +151,124 @@
+ *   specify the identity and choice of a voter. The voteCounter is making that
+ *   available to the Registrar, which should wrap and attenuate it so each
+ *   voter gets only the ability to cast their own vote at a weight specified by
+ *   the registrar.
+ * @property {SubmitVote} submitVote
  */
 
 /**
- * @typedef {Object} BallotCounterPublicFacet
+ * @typedef {Object} VoteCounterPublicFacet
  * @property {() => boolean} isOpen
- * @property {() => Ballot} getBallotTemplate
- * @property {() => Promise<string>} getOutcome
+ * @property {() => Question} getQuestion
+ * @property {() => Promise<Position>} getOutcome
+ * @property {() => QuestionDetails} getDetails
  * @property {() => Promise<VoteStatistics>} getStats
  */
 
 /**
- * @typedef {Object} BallotCounterCloseFacet
+ * @typedef {Object} VoteCounterCloseFacet
  *   TEST ONLY: Should not be allowed to escape from contracts
  * @property {() => void} closeVoting
  */
 
 /**
- * @typedef {Object} CompleteEqualWeightBallot
- * @property {string} question
- * @property {string[]} chosen - a list of equal-weight preferred positions
+ * @typedef {Object} VoteCounterFacets
+ * @property {VoteCounterPublicFacet} publicFacet
+ * @property {VoteCounterCreatorFacet} creatorFacet
+ * @property {VoteCounterCloseFacet} closeFacet
  */
 
 /**
- * @typedef {Object} CompleteWeightedBallot
- * @property {string} question
- * @property {Record<string,bigint>[]} weighted - list of positions with weights.
- *   BallotCounter may limit weights to a range or require uniqueness.
+ * @callback BuildVoteCounter
+ * @param {QuestionSpec} questionSpec
+ * @param {bigint} threshold - questionSpec includes quorumRule; the registrar
+ *    converts that to a number that the counter can enforce.
+ * @param {Instance} instance
+ * @returns {VoteCounterFacets}
  */
 
 /**
- * @typedef {Object} CompleteOrderedBallot
- * @property {string} question
- * @property {string[]} ordered - ordered list of position from most preferred to
- *   least preferred
+ * @callback LooksLikeQuestionSpec
+ * @param {unknown} allegedQuestionSpec
+ * @returns {QuestionSpec}
+ */
+
+/**
+ * @callback LooksLikeParamChangeIssue
+ * @param {unknown} issue
+ * @returns { asserts issue is ParamChangeIssue }
+ */
+
+/**
+ * @callback LooksLikeIssueForType
+ * @param {ElectionType} electionType
+ * @param {unknown} issue
+ * @returns { asserts issue is Issue }
+ */
+
+/**
+ * @callback LooksLikeSimpleIssue
+ * @param {unknown} issue
+ * @returns { asserts issue is SimpleIssue }
  */
 
 /**
- * @typedef { CompleteEqualWeightBallot | CompleteOrderedBallot | CompleteWeightedBallot } CompletedBallot
+ * @callback LooksLikeClosingRule
+ * @param {unknown} closingRule
+ * @returns { asserts closingRule is ClosingRule }
  */
 
 /**
  * @callback SubmitVote
- * @param {Handle<'Voter'>} seat
- * @param {ERef<CompletedBallot>} filledBallot
+ * @param {Handle<'Voter'>} voterHandle
+ * @param {[Position]} chosenPositions
```

Comment:
`Array<Position>` can be abbreviated as `Position[]` but not as `[Position]`.

```suggestion
 * @param {Position[]} chosenPositions
```

Reply:
done